Add docs for kit library (#1213)
onflow/docs: docs/tools/kit/index.md

---
title: "@onflow/kit"
description: React hooks for interacting with the Flow blockchain.
sidebar_position: 2
---

# @onflow/kit

> > 🚧 This library is currently in alpha and is subject to change.

Reviewer: ```suggestion
:::warning

🚧 This library is currently in alpha and is subject to change.

:::
```

> `@onflow/kit` is a lightweight React utility library that simplifies interacting with the Flow blockchain. It provides a collection of hooks and (soon) components designed to make authentication, script execution, transactions, event subscriptions, and network configuration seamless in React apps.

Reviewer: ```suggestion
`@onflow/kit` is a lightweight React utility library that simplifies interacting with the Flow blockchain. It provides a collection of hooks, similar to those in other popular web3 libraries, that make it easier to build frontends that understand blockchain interactions.  **In the future**, it will also provided components designed to make authentication, script execution, transactions, event subscriptions, and network configuration seamless in React apps.
```

## 🔌 Included React Hooks

- [`useCurrentFlowUser`](#usecurrentflowuser) – Authenticate and manage the current Flow user
- [`useFlowAccount`](#useflowaccount) – Fetch Flow account details by address
- [`useFlowBlock`](#useflowblock) – Query latest or specific Flow blocks
- [`useFlowConfig`](#useflowconfig) – Access the current Flow configuration
- [`useFlowEvents`](#useflowevents) – Subscribe to Flow events in real-time
- [`useFlowQuery`](#useflowquery) – Execute Cadence scripts with optional arguments
- [`useFlowMutate`](#useflowmutate) – Send transactions to the Flow blockchain
- [`useFlowTransaction`](#useflowtransaction) – Track transaction status updates

> 
> 

Reviewer: ```suggestion

```

## Installation

```bash
npm install @onflow/kit
```

## Usage

### Wrapping Your App With `FlowProvider`

Before using any hooks or components (coming soon), wrap your application with the `FlowProvider` to initialize FCL configuration. This sets up FCL and maps its configuration keys to a strictly typed format for your hooks.

> ```jsx

Reviewer: ```suggestion
```tsx
```

import React from "react"
import App from "./App"
import { FlowProvider } from "@onflow/kit"
import flowJSON from "../flow.json"

function Root() {
  return (
    <FlowProvider
      config={{
        accessNodeUrl: "https://access-mainnet.onflow.org",
        flowNetwork: "mainnet",
        appDetailTitle: "My On Chain App",
        appDetailIcon: "https://example.com/icon.png",
        appDetailDescription: "A decentralized app on Flow",
        appDetailUrl: "https://myonchainapp.com",
        // include other typed configuration keys as needed...
      }}
      flowJson={flowJSON}
    >
      <App />
    </FlowProvider>
  )
}

export default Root
```

> 

Reviewer: ```suggestion

If you're using [Next.js], put this in `layout.tsx`.  Adapt as appropriate for other frontend frameworks.

```

  const sendTransaction = () => {
    mutate({
      cadence: `transaction() {
        prepare(acct: &Account) {
          log(acct.address)
        }
      }`,
      args: (arg, t) => [],
      proposer: fcl.currentUser,
      payer: fcl.currentUser,
      authorizations: [],
      limit: 100,
    })
  }

  return (
    <div>
      <button onClick={sendTransaction} disabled={isPending}>
        Send Transaction
      </button>
      {isPending && <p>Sending transaction...</p>}
      {error && <p>Error: {error.message}</p>}
      {txId && <p>Transaction ID: {txId}</p>}
    </div>
  )
}
```

---

### `useFlowTransaction`

```tsx
import { useFlowTransaction } from "@onflow/kit"
```

#### Parameters:

- `txId: string` – Transaction ID to subscribe to

#### Returns:

- `transactionStatus: TransactionStatus | null`
- `error: Error | null`

```tsx
function TransactionComponent() {
  const txId = "your-transaction-id-here"
  const { transactionStatus, error } = useFlowTransaction(txId)

  if (error) return <div>Error: {error.message}</div>

  return (
    <div>
      Status: {transactionStatus?.statusString}
    </div>
  )

> }

Reviewer: ```suggestion
}

[Next.js]: https://nextjs.org/
```
